src/wildcard.jl

```diff
@@ -0,0 +1,29 @@
+struct WildcardDimensions{R} <: AbstractDimensions{R}
+    # We simply choose Bool since it will get promoted
+    WildcardDimensions() = new{Bool}()
+end
+
+constructorof(::Type{<:WildcardDimensions}) = WildcardDimensions
+dimension_names(::Type{<:WildcardDimensions}) = ()
+
+# Wildcard dimensions are always compatible:
+Base.:(==)(::AbstractDimensions, ::WildcardDimensions) = true
+Base.:(==)(::WildcardDimensions, ::AbstractDimensions) = true
+
+# For any zero/oneunit specified on the type, we return a wildcard dimension instead:
+Base.zero(::Type{Q}) where {T,Q<:UnionAbstractQuantity{T}} = constructorof(Q)(zero(T), WildcardDimensions())
+
+# For propagation rules, we assume all relevant dimensions are zero
+Base.getproperty(::WildcardDimensions, ::Symbol) = zero(Bool)
+Base.iszero(::WildcardDimensions) = true
+
+# Always promote to the other dimension type:
+Base.promote_rule(::Type{D}, ::Type{<:WildcardDimensions}) where {D<:AbstractDimensions} = D
+
+# Other utility rules:
+Base.show(io::IO, ::WildcardDimensions) = print(io, "[*]")
```

Comment:
If `dimension(x)` is printed with `x` having `WildcardDimensions`, this looks like an array (totally unimportant since this is WIP, but just writing before I forget)